<?xml version="1.0"?>
<documentation xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:noNamespaceSchemaLocation="https://phpcsstandards.github.io/PHPCSDevTools/phpcsdocs.xsd"
    title="Require Exit/Die Parentheses"
    >
    <standard>
    <![CDATA[
    Require the use of parentheses when calling `exit` or `die`, even if no argument is given.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Calling exit/die with parentheses.">
        <![CDATA[
die<em>()</em>;
exit<em>($status)</em>;
        ]]>
        </code>
        <code title="Invalid: Calling exit/die without parentheses.">
        <![CDATA[
die<em></em>;
exit<em></em>;
        ]]>
        </code>
    </code_comparison>
</documentation>
