<?xml version="1.0"?>
<documentation xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:noNamespaceSchemaLocation="https://phpcsstandards.github.io/PHPCSDevTools/phpcsdocs.xsd"
    title="Disallow Attribute Parentheses"
    >
    <standard>
    <![CDATA[
    Disallows the use of parentheses when instantiating an attribute without passing parameters.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Attribute instantiation without parentheses or with parameters.">
        <![CDATA[
#[MyAttribute<em></em>]
#[MyAttribute<em>(self::BAZ)</em>]
class Foo {}
        ]]>
        </code>
        <code title="Invalid: Attribute instantiation with parentheses, but without parameters.">
        <![CDATA[
#[MyAttribute<em>()</em>]
class Foo {}
        ]]>
        </code>
    </code_comparison>
</documentation>
